documentation/status-proxy-service.md

```diff
@@ -0,0 +1,123 @@
+to the real `status-service`. 
+
+This could be useful for instance, if you have a load balancer that needs to
+access the `status-service`, but which cannot be configured to use custom SSL
+certs from pem files.
+
+## Preferred Alternatives
+Given that the `status-proxy-service` opens a small hole in your security,
+consider the following alternatives first:
+
+* Configure your load balancer to use HTTPS with pem files if possible
+* If you are comfortable setting up reverse proxying with a tool such as nginx,
+  consider using that
+* Consider running the regular status service on a plaintext port, so that only
+  the status service is accessible via plaintext, and you are not reverse
+  proxying any requests to the HTTPS portions of the apps
+
+## Security Considerations
+There is some security risk associated with providing a plaintext window into
+a portion of your HTTPS endpoints.
+  
+* If incorrectly configured, the `status-proxy-service` could unintentionally
+  open up endpoints other than the `status-service` to plaintext access
+* The proxy should be run on a network interface that is not accessible from
+  outside your internal LAN if possible
+* The proxy should be run on a port that is firewalled to only allow access
+  from your load balancer
+
+## Examples
+
+### Configuring the status-service for plaintext access
+
+If you would like to configure your status service to be on a plaintext port,
+your trapperkeeper configuration might look something like this:
+
+```
+webserver: {
+  default: {
+    ssl-port: 9001
+    ssl-cert: ./ssl/certs/localhost.pem
+    ssl-key: ./ssl/private_keys/localhost.pem
+    ssl-ca-cert: ./ssl/certs/ca.pem
+    default-server: true
+  }
+
+  status: {
+    port: 8080
+  }
+}
+
+web-router-service: {
+    "puppetlabs.trapperkeeper.services.status.status-service/status-service": {
+      route: /status
+      server: status
+    }
+}
+```
+
+This config contains a new jetty server called `status` running on port 8080
+that is separate from the other, SSL enabled, server. In the 
+`web-router-service` section, the status service is configured to run on the
+`status` server, and be mounted at `/status`
+
+
+### Configuring the status-proxy-service
+
+If you would like to setup the status proxy service, your trapperkeeper 
+configuration might look like this:
+
+```
+webserver: {
+  default: {
+    ssl-port: 9001
+    ssl-cert: ./ssl/certs/localhost.pem
+    ssl-key: ./ssl/private_keys/localhost.pem
+    ssl-ca-cert: ./ssl/certs/ca.pem
+    default-server: true
+  }
+
+  status-proxy: {
+    port: 8080
+  }
+}
+
+web-router-service: {
+    "puppetlabs.trapperkeeper.services.status.status-service/status-service": {
+      route: /status
+      server: default
+    }
+
+    "puppetlabs.trapperkeeper.services.status.status-proxy-service/status-proxy-service": {
+      route: /status-proxy
+      server: status-proxy
+    }
+}
+
+status-proxy: {
+  proxy-target-url: "https://localhost:9001/status"
```

Comment:
Can't remember the answer to this offhand, but it may be the case that the Jetty http proxy client would validate that the hostname used in the target URL matches the CN or a SubjectAltName in the server certificate - the server certificate presented by the Jetty server running at port 9001 in this example.  If that is the case, probably worth mentioning that the hostname in the `proxy-target-url` must match the CN or a SubjectAltName in the server certificate since "localhost" probably wouldn't be on the certificate used in production even though a host name of "localhost" would likely still be resolvable from the machine doing the proxying.  Maybe would be good to use something other than "localhost" throughout these examples to make that distinction more clear, e.g., "myhostname".